torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -849,8 +851,16 @@ def _run_param_init_fn():
             self._pre_forward_handles: List[RemovableHandle] = []
             self._post_forward_handles: List[RemovableHandle] = []
             self._param_exec_order_state = ParamExecOrderState.UNINITIALIZED
+            # A list that stores the handles based on the parameter execution order
+            self._handles_exec_order: List[FlatParamHandle] = []
             handle_init_mode = auto_wrap_policy.handle_init_mode
+            # TODO (linjianma): introduce module level group_policy
             self._register_param_handles_from_root_module(module, handle_init_mode)
+            # self.flatparam_handle_dict maps param to handle, it will be used to
+            # get handle execution ordering.
+            self.flatparam_handle_dict: Dict[FlatParameter, FlatParamHandle] = dict()
```

Comment:
Can we rename this to `flat_param_to_handle` since that is the naming convention I typically follow (i.e. `key_to_value`)? 😅 
- I like this convention since typically it makes it clear from the name what are the keys and what the values. 
- I generally avoid including the data structure in the name in case we do change it. For a `dict`, it may not be as likely, but for other cases it may happen. (E.g. we may change a list to a set, or we may have a logical mapping that is implemented as a `Dict[int, Any]` but change it to a `List[Any]` since the keys are `0`, ..., `n-1`.)

I also usually call everything `handle` instead of `flatparam_handle` when unambiguous because in the future, I may extend to other kinds of handles. For example, to support multiple parameter groups within one `FlatParameter`, I was considering having `OrigParamFlatParamHandle` (or some other name) that inherits from `FlatParamHandle`. For a different sharding strategy, we may have a handle that does not manage a `FlatParameter`.

Reply:
done!